test/saithrift/tests/sai_base_test.py

```diff
@@ -39,6 +42,30 @@ def setUp(self):
         else:
             server = 'localhost'
 
+        if self.test_params.has_key("port_map"):
+            user_input = self.test_params['port_map']
+            splitted_map = user_input.split(",")
+            for item in splitted_map:
+                interface_front_pair = item.split("@")
+                interface_to_front_mapping[interface_front_pair[0]] = interface_front_pair[1]
+        elif self.test_params.has_key("port_map_file"):
+            user_input = self.test_params['port_map_file']
+            f = open(user_input, 'r')
+            for line in f:
+                if (len(line) > 0 and (line[0] == '#' or line[0] == ';' or line[0]=='/')):
+                    continue;
+                interface_front_pair = line.split("@")
+                interface_to_front_mapping[interface_front_pair[0]] = interface_front_pair[1].strip()
+        else:
+            f = open('./tests/default_interface_to_front_map.ini', 'r')
+            for line in f:
+                if (len(line) > 0 and (line[0] == '#' or line[0] == ';' or line[0]=='/')):
+                    continue;
+                interface_front_pair = line.split("@")
+                interface_to_front_mapping[interface_front_pair[0]] = interface_front_pair[1].strip()		    
```

Comment:
given that we change the interface map file format, I think it's better to make the map file as a mandatory input file.